pkg/api/error/error.go

```diff
@@ -61,6 +61,8 @@ func (e *apiError) statusCode() int {
 		return http.StatusRequestEntityTooLarge
 	case TypeNotAcceptable:
 		return http.StatusNotAcceptable
+	case TypeUnavailable:
```

Comment:
@56quarters are we sure that this will not introduce any side-effect to the circuit breaker?

Reply:
Circuit breakers work entirely on turning errors into gRPC codes: https://github.com/grafana/mimir/blob/da98104bb2c606cf4d0c8fd303d73aa82f77b5d8/pkg/ingester/client/circuitbreaker.go#L94-L104